hedwig.py

```diff
@@ -0,0 +1,344 @@
+        try:
+            volume = data['volume']
+            article.volume = volume
+        except KeyError as e:
+            pass
+
+        try:
+            year = data['year']
+            article.publication_year = year
+        except KeyError as e:
+            pass
+
+        response = queries.make_academic_article.run(connection, **params)
+        print(response)
+        if response:
+            ARTICLE_COUNT += 1     
+        
+    add_authors(connection, article, data)
+
+def match_input(connection, label, obj_type):
+    #TODO: special match function for doi
+    deets = queries.find_n_for_label.get_params(connection)
+    deets['Thing'].name = label
+    deets['Thing'].category = obj_type
+
+    current_list = queries.find_n_for_label.run(connection, **deets)
+    choices = {}
+    for key, val in current_list.items():
+        if label.lower() == val.lower():
+            choices[key] = val
+
+    if len(choices) == 1:
+        for key in choices:
+            return key
+    else:
+        #TODO: deal with duplicates
+        return None
+
+def check_filter(label, category):
+    cleanfig_path = os.path.join("filters", "extras.yaml")
+    with open(cleanfig_path, 'r') as config_file:
+        cleanfig = yaml.load(config_file.read())
+    #primary cleaning
+    abbrev_table = cleanfig.get('abbrev_table')
+    label += " " #add trailing space in case abbreviation is end of string
+    label = label.replace("  ", " ")
+    label = label.replace("\\", "") #can not have escape character when using update api
+    for abbrev in abbrev_table:
+        label = label.replace(abbrev, abbrev_table[abbrev])
+    if label[-1] == " ":
+        label = label[:-1] #remove final space
+
+    #match against filter
+    filter_file = category + "_filter.txt"
+    filter_path = os.path.join("filters", filter_file)
+    pairs = {}
+    with open(filter_path) as fil:
+        content = fil.readlines()
+        for line in content:
+            key, val = line.split("|")
+            pairs[key] = val
+
+        for key, val in pairs.items():
+            if label == key:
+                label = val
+                break
+    
+    #return clean_label
+    return label
+
+def create_journal(connection, journal_name, publisher_name):
+    global PUBLISHER_COUNT
+    global JOURNAL_COUNT
+    clean_pname = check_filter(publisher_name.title(), 'publisher')
+    publisher_n = match_input(connection, clean_pname, 'publisher')
+
+    reqs = queries.make_journal.get_params(connection)
+    reqs['Journal'].name = journal_name
+    if publisher_n:
+        reqs['Publisher'].n_number = publisher_n
+    else:
+        reqs['Publisher'].name = clean_pname
+        PUBLISHER_COUNT += 1
+
+    response = queries.make_journal.run(connection, **reqs)
+    print(response)
+    if response:
+        JOURNAL_COUNT += 1
+        return reqs['Journal'].n_number
+    else:
+        return None
+
+def add_authors(connection, article, data):
+    global AUTHOR_COUNT
```

Comment:
second time the global is declared in the file. Does this work? Try to avoid global usage

Reply:
For instance, this function could return the number of authors added or something instead